pyproject.toml

```
@@ -108,10 +110,34 @@ pyupgrade = { version = "3.16.0", optional = true }
 # numpy = [
 # 	{version="1.24.4", python = "<3.9", optional = true},
 # 	{version="1.26.4", python = ">=3.9", optional = true}]
+ruff = "^0.7.1"
```

shall we har dpin like everything else for consistency?  pending some bugs github needs to address, we plan on changing this pattern for many dependencies.

```suggestion
ruff = "0.7.1"
```

---

Taken